addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js

```diff
@@ -207,7 +209,11 @@ export class Powerbox {
                 true,
             );
             this._lastText = diff.bMove.join('');
-            if (this._lastText.match(/\s/) && this._currentOpenOptions.closeOnSpace !== false) {
+            const selection = this.options.document.getSelection();
+            if (
+                (this._lastText.match(/\s/) && this._currentOpenOptions.closeOnSpace !== false) ||
+                initialTarget !== closestBlock(selection.anchorNode)
```

Comment:
Same here:
```suggestion
                !selection ||
                initialTarget !== closestBlock(selection.anchorNode)
```
This is a bit out of scope though, it's not part of the fix. Please make a new commit for it, along with its test.